### DISCUSSION TITLE
My Sql - Calling Stored Procedures w/ Parameter Values That Contain Null

### DISCUSSION THREAD

**Discussion by chadgriff** [Budibase/budibase#3112] [Help] [answered]
I am attempting to call a stored procedure (MS SQL) from the front end of the application.  There are 4 parameters I pass to the stored procedure.  The SP can accommodate parameters which contain either NULL or ACTUAL values.  However, I am getting a Budibase error when calling the SP if I don't populate all four parameters with ACTUAL values.  One param is a record id so it will never be null. However, the remaining 3 params are being populated with an Option Picker and any one (or all of them) may be NULL (i.e. no choice selected by the user).  As I stated earlier, if I make a selection in all 3 option pickers, the call to the SP goes fine.  BUT, if I leave any of the 3 blank (no choice selected)...then Budibase causes an error and doesn't make the call out to the SP.  Any guidance or tips to get around this would be appreciated!
![sp_null_1](https://user-images.githubusercontent.com/15015476/138362568-a56b0c25-606a-4fb7-95fe-3fa87aaaf786.png)
![sp_null_2](https://user-images.githubusercontent.com/15015476/138362638-4984394a-6d9d-4e3f-8605-153d23d40ef6.png)

**Comment by mjashanks** [comment, marked as answer]
Hi Chad.

In you parameter value, you'll need to use an `#if`. Mine looks like:

`{{ #if  New Form.Fields.possiblyNull }}{{ New Form.Fields.possiblyNull }}{{else}}NULL{{/if}}`

I appreciate that this is quite hard to read (and write). Syntax can be found here: https://handlebarsjs.com/guide/builtin-helpers.html#if

This is about to get a lot easier - we are adding javascript support in a few weeks.

**Reply by chadgriff** [reply]
Thanks Michael...that did the trick!  I will admit that learning the handlebars syntax has been one of the largest learning curves for me personally as part of my Budibase journey to date. In all transparency though, I am not very proficient in HTML, Javascript, etc. so that may be setting me back a bit when it comes to handlebars syntax.  Budibase is great though...really a strong addition to my toolbox!
